README.md

```diff
@@ -0,0 +1,136 @@
+# State Ex Machina
+
+State Ex Machina is a MVI-like library written in Kotlin for Android.
+
+**Why do we need another MVI library?**\
+Because we simply couldn't find one that was easy to start working with, lightweight and that would cover enough use cases.
+
+**Core concepts**
+  - state machine built on kotlin ```Flow``` to handle and store state changes  
+  - reactive state that can be applied to the UI 
+  - declarative model definition, how and why?
+  - clear DSL
+  - lightweight
+
+## Getting started
+
+Here's a simple example to show off the fundaments on which the library is based on.\ 
+The user wants to insert a number, sum it with the previous inserted number (initial value is zero) and see the result.\
+Lastly saving the total sum with a network call.
+
+### Define the dependencies
+
+```kotlin
+// MVI foundation
+implementation("com.state-ex-machina:core:<latest-version>")
+// Jetpack Compose MVI extensions
+implementation("com.state-ex-machina:ext-compose:<latest-version>")
+// Android ViewModel MVI extensions
+implementation("com.state-ex-machina:ext-view-model:<latest-version>")
+```
+
+### Define the Intents
+
+Intents represent user intentions: the actions with which the user achieves an objective
+
+```kotlin
+sealed interface CounterIntent : Intent {
+    data class TypeNumber(val number: Int) : CounterIntent
+    object Count : CounterIntent
+    object SaveTotal : CounterIntent
+}
+```
+
+### Define the State
+
+State represents the current condition of the data and UI elements 
```

Comment:
Mmmmh... non credo si capisca molto bene. Forse più qualcosa del tipo "State represents a photo of all the dynamic information needed to render the view and to let the model interact with the domain (and update itself)"